test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs

```diff
@@ -0,0 +1,609 @@
+        }
+
+        [Fact]
+        public async Task CanSendDocumentSymbolRequest()
+        {
+            string scriptPath = NewTestFile(@"
+function CanSendDocumentSymbolRequest {
+
+}
+
+CanSendDocumentSymbolRequest
+");
+
+            SymbolInformationOrDocumentSymbolContainer symbolInformationOrDocumentSymbols =
+                await LanguageClient.SendRequest<SymbolInformationOrDocumentSymbolContainer>(
+                    "textDocument/documentSymbol",
+                    new DocumentSymbolParams
+                    {
+                        TextDocument = new TextDocumentIdentifier
+                        {
+                            Uri = new Uri(scriptPath)
+                        }
+                    });
+
+            Assert.Collection(symbolInformationOrDocumentSymbols,
+                symInfoOrDocSym => {
+                    Range range = symInfoOrDocSym.SymbolInformation.Location.Range;
+
+                    Assert.Equal(1, range.Start.Line);
+                    Assert.Equal(0, range.Start.Character);
+                    Assert.Equal(3, range.End.Line);
+                    Assert.Equal(1, range.End.Character);
+                });
+        }
+
+        [Fact]
+        public async Task CanSendReferencesRequest()
+        {
+            string scriptPath = NewTestFile(@"
+function CanSendReferencesRequest {
+
+}
+
+CanSendReferencesRequest
+");
+
+            LocationContainer locations = await LanguageClient.SendRequest<LocationContainer>(
+                "textDocument/references",
+                new ReferenceParams
+                {
+                    TextDocument = new TextDocumentIdentifier
+                    {
+                        Uri = new Uri(scriptPath)
+                    },
+                    Position = new Position
+                    {
+                        Line = 5,
+                        Character = 0
+                    },
+                    Context = new ReferenceContext
+                    {
+                        IncludeDeclaration = false
+                    }
+                });
+
+            Assert.Collection(locations,
+                location1 =>
+                {
+                    Range range = location1.Range;
+                    Assert.Equal(1, range.Start.Line);
+                    Assert.Equal(9, range.Start.Character);
+                    Assert.Equal(1, range.End.Line);
+                    Assert.Equal(33, range.End.Character);
+
+                },
+                location2 =>
+                {
+                    Range range = location2.Range;
+                    Assert.Equal(5, range.Start.Line);
+                    Assert.Equal(0, range.Start.Character);
+                    Assert.Equal(5, range.End.Line);
+                    Assert.Equal(24, range.End.Character);
+                });
+        }
+
+        [Fact(Skip = "Bug in Newtonsoft that csharp-language-server-protocol will workaround")]
```

Comment:
Possible to explain what the issue is and/or link to the issue here in a comment?

Reply:
They've fixed the issue :) and the tests are passing now!